pkg/types/azure/platform.go

```diff
@@ -100,6 +102,25 @@ type Platform struct {
 
 	// CustomerManagedKey has the keys needed to encrypt the storage account.
 	CustomerManagedKey *CustomerManagedKey `json:"customerManagedKey,omitempty"`
+
+	// BootDiagnostics has the value for the storage account URI where the
+	// machine log information is sent to. Defaults to disabled if no value
+	// and defaults to the installer created storage account if the URI is not
+	// mentioned.
+	// +optional
+	BootDiagnostics *BootDiagnostics `json:"bootDiagnostics,omitempty"`
+}
+
+type BootDiagnostics struct {
+	// Type specifies the boot diagnostics type for the machines created.
+	//
+	// +kubebuilder:validation:Enum=Disabled;Managed;UserManaged
+	// +kubebuilder:default=Disabled
+	Type capz.BootDiagnosticsStorageAccountType `json:"type"`
+
+	// StorageAccountURI specifies the storage account the log information
+	// needs to be stored in.
+	StorageAccountURI string `json:"storageAccountURI"`
```

Comment:
Perhaps this should just be the name of the storage account. Default to using the storage account the cluster creates if omitted. You can build the URI like we do here: 

https://github.com/openshift/installer/blob/694d083d3332bd5b892f0098a91be6c206b18fce/pkg/infrastructure/azure/azure.go#L339-L340